kardia/params.go

```diff
@@ -180,23 +190,37 @@ func (ec *Client) GetTotalProposals(ctx context.Context) (*big.Int, error) {
 // GetProposals returns list of proposals
 func (ec *Client) GetProposals(ctx context.Context, pagination *types.Pagination) ([]*types.ProposalDetail, uint64, error) {
 	one := big.NewInt(1)
-	start := new(big.Int).SetInt64(int64(pagination.Skip))
-	end := new(big.Int).SetInt64(int64(pagination.Limit))
 	total, err := ec.GetTotalProposals(ctx)
 	if err != nil {
 		return nil, 0, err
 	}
-	if end.Cmp(total) == 1 {
-		end = total
+	var (
+		start = new(big.Int).SetInt64(0)
+		end   = total
+	)
+	if pagination != nil {
+		start = new(big.Int).SetInt64(int64(pagination.Skip))
+		end = new(big.Int).SetInt64(int64(pagination.Limit))
+		if end.Cmp(total) == 1 {
+			end = total
+		}
 	}
 	var result []*types.ProposalDetail
 	// i must be a new int so that it does not overwrite start
 	for i := new(big.Int).Set(start); i.Cmp(end) < 0; i.Add(i, one) {
-		metadata, err := ec.GetProposalDetails(ctx, i)
+		detail, err := ec.GetProposalDetails(ctx, i)
 		if err != nil {
 			continue
 		}
-		result = append(result, metadata)
+		result = append(result, detail)
 	}
 	return result, total.Uint64(), nil
 }
+
+func ConvertNetworkParamValue(fieldName string, value *big.Int) interface{} {
+	if cfg.ParamKeysTypeNumber[fieldName] {
+		return value.Uint64()
+	} else {
```

Comment:
🚫 **[golangci]** <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>`if` block ends with a `return` statement, so drop this `else` and outdent its block (golint)